GEOSldas_App/util/postproc/ObsFcstAna_stats/helper/compute_monthly_stats.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import sys
+import os
+sys.path.append('../../../shared/python/')
+
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from read_GEOSldas import read_ObsFcstAna, read_tilecoord, read_obs_param
+
+def compute_monthly_stats(expdir_list,expid_list,domain,this_month,tc,obsparam_list,var_list,obs_from):
+
+    n_tile = tc['N_tile']
+    n_spec = len(obsparam_list[0])
+
+    #this_month = [start_date_ofthemonth, end_date_ofthemonth]
+    start_time = this_month[0]
+    end_time = this_month[1]
+    
+    data_sum = {}
+    data2_sum = {}
+
+    N_data = np.zeros((n_tile, n_spec))
+    oxf_sum = np.zeros((n_tile, n_spec))
+    oxa_sum = np.zeros((n_tile, n_spec))
+    fxa_sum = np.zeros((n_tile, n_spec))
+
+    for var in var_list:
+        data_sum[var] = np.zeros((n_tile, n_spec))
+        data2_sum[var] = np.zeros((n_tile, n_spec))
+
+    date_time = start_time
+    while date_time < end_time:
+        
+        # read the list of experiments at each time step
+        OFA_list =[]
+        for i in range(len(expdir_list)):
+            fname = expdir_list[i]+expid_list[i]+'/output/'+domain+'/ana/ens_avg/Y'+ \
+                              date_time.strftime('%Y') + '/M' + \
+                              date_time.strftime('%m') + '/'  + \
+                              expid_list[i]+'.ens_avg.ldas_ObsFcstAna.' + \
+                              date_time.strftime('%Y%m%d_%H%M') +'z.bin'
+
+            if os.path.isfile(fname):
+                print('read '+fname)
+                OFA_list.append(read_ObsFcstAna(fname))
+
+        data_all=[]
+        for OFA, obs_param in zip(OFA_list,obsparam_list):
+
+            # Initialize full size variable for an experiment
+            data_tile={}
+            for var in var_list:
+                data_tile[var] = np.zeros((n_tile, n_spec)) +np.nan
+
+            if len(OFA['obs_tilenum']) > 0:
+                for ispec in np.arange(n_spec):
+                    # check species overall "assim" flag for masking
+                    this_species = int(obs_param[ispec]['species'])
+                    masked_data = {}
+
+                    # get mask 
+                    if obs_param[ispec]['assim'] == 'T':
+                        is_valid = np.logical_and(OFA['obs_species'] == this_species, OFA['obs_assim']==1)
+                    else:
+                        is_valid = OFA['obs_species'] == this_species
+
+                    masked_tilenum = OFA['obs_tilenum'][is_valid]
+                    for var in var_list:
+                        masked_data[var] = OFA[var][is_valid]
+          
+                    tile_idx = np.where(np.isin(tc['tile_id'], masked_tilenum))[0]
```

Comment:
I think we need something like the following to deal with regional runs when `OFA['obs_tilenum']` and `tc['tile_id']` will no longer align:

 ```
                # Convert to 0-based indices for tc array
                tile_indices = masked_tilenum - 1
        
                # Get corresponding tc tile IDs
                tc_tile_ids = tc['tile_id'][tile_indices]

                tile_idx = np.where(np.isin(tc['tile_id'], tc_tile_ids))[0]
```

Reply:
Seem we don't need 'tile_id' here, does it make sense to change L67 to tile_idx = OFA['obs_tilenum'][is_valid], and remove L71?

Reply:
correction,      tile_idx = OFA['obs_tilenum'][is_valid] - 1 to make it 0-based

Reply:
Yes, I think `tile_idx = OFA['obs_tilenum'][is_valid] - 1 ` is correct. The actual 'tile_id' isn't required here.